### DISCUSSION TITLE
Tendermint on old CPUs

### DISCUSSION THREAD

**Discussion by smalla** [tendermint/tendermint#9662] [General]
is tendermint supposed to run on old cpus like E5-4650? I havent found in any documentation if there is a limitation, but Im getting this error: 
(*BlockchainReactor).poolRoutine\n\tgithub.com/tendermint/tendermint@v0.34.14/blockchain/v0/reactor.go:398\nfailed to execute message; message index: 0: Error calling the VM: Error instantiating a Wasm module: Error instantiating module: CpuFeature("EnumSet(BMI1 | BMI2 | AVX2 | LZCNT)"): execute wasm contract failed" module=state

my CPU does not have these features, and Im not sure if it is a feature or a bug.
(or it is a limitation of wasm eventually?)

Thank You.

**Comment by thanethomson** [comment]
Tendermint doesn't use WASM internally for anything. Probably a good idea to speak to whoever maintains the application you're running on top of Tendermint.

**Reply by smalla** [reply]
so it supposed to run on every processor and this particular errors is coming from in the wasm module ?

**Reply by thanethomson** [reply]
We make no guarantees about Tendermint running on every processor. Also, this error is not coming from Tendermint at all. Tendermint doesn't have a WASM VM inside of it. What application are you running on top of Tendermint?

**Reply by smalla** [reply]
It is a terrad daemon, supposed to be a validator node. I have replaced the server since last message, so cannot test it anymore.
However I have found out this only happens if there are invalid transactions to be processed.  Thank You for Your time.

**Reply by thanethomson** [reply]
Transaction processing is handled by the application (in this case that's terrad). You are encountering an application problem here - not a Tendermint one.
